core/src/avm2/globals/flash/utils/bytearray.rs

```diff
@@ -0,0 +1,946 @@
+    }
+    Ok(Value::Undefined)
+}
+
+pub fn clear<'gc>(
+    activation: &mut Activation<'_, 'gc, '_>,
+    this: Option<Object<'gc>>,
+    _args: &[Value<'gc>],
+) -> Result<Value<'gc>, Error> {
+    if let Some(mut bytearray) = this
+        .unwrap()
+        .as_bytearray_mut(activation.context.gc_context)
+    {
+        bytearray.clear();
+    }
+    Ok(Value::Undefined)
+}
+
+pub fn position<'gc>(
+    _activation: &mut Activation<'_, 'gc, '_>,
+    this: Option<Object<'gc>>,
+    _args: &[Value<'gc>],
+) -> Result<Value<'gc>, Error> {
+    if let Some(bytearray) = this.unwrap().as_bytearray() {
+        return Ok(Value::Unsigned(bytearray.position() as u32));
+    }
+    Ok(Value::Undefined)
+}
+
+pub fn set_position<'gc>(
+    activation: &mut Activation<'_, 'gc, '_>,
+    this: Option<Object<'gc>>,
+    args: &[Value<'gc>],
+) -> Result<Value<'gc>, Error> {
+    if let Some(mut bytearray) = this
+        .unwrap()
+        .as_bytearray_mut(activation.context.gc_context)
+    {
+        let num = args
+            .get(0)
+            .unwrap_or(&Value::Integer(0))
+            .coerce_to_u32(activation)?;
+        bytearray.set_position(num as usize);
+    }
+    Ok(Value::Undefined)
+}
+
+pub fn bytes_available<'gc>(
+    _activation: &mut Activation<'_, 'gc, '_>,
+    this: Option<Object<'gc>>,
+    _args: &[Value<'gc>],
+) -> Result<Value<'gc>, Error> {
+    if let Some(bytearray) = this.unwrap().as_bytearray() {
+        return Ok(Value::Unsigned(
+            if bytearray.position() > bytearray.bytes().len() {
+                0
+            } else {
+                (bytearray.bytes().len() - bytearray.position()) as u32
+            },
+        ));
+    }
+    Ok(Value::Undefined)
+}
+
+pub fn length<'gc>(
+    _activation: &mut Activation<'_, 'gc, '_>,
+    this: Option<Object<'gc>>,
+    _args: &[Value<'gc>],
+) -> Result<Value<'gc>, Error> {
+    if let Some(bytearray) = this.unwrap().as_bytearray() {
+        return Ok(Value::Unsigned(bytearray.bytes().len() as u32));
+    }
+    Ok(Value::Undefined)
+}
+
+pub fn set_length<'gc>(
+    activation: &mut Activation<'_, 'gc, '_>,
+    this: Option<Object<'gc>>,
+    args: &[Value<'gc>],
+) -> Result<Value<'gc>, Error> {
+    if let Some(mut bytearray) = this
+        .unwrap()
+        .as_bytearray_mut(activation.context.gc_context)
+    {
+        let len = args
+            .get(0)
+            .unwrap_or(&Value::Unsigned(0))
+            .coerce_to_u32(activation)
+            .unwrap() as usize;
+        bytearray.resize(len);
+    }
+    Ok(Value::Undefined)
+}
+
+pub fn endian<'gc>(
+    activation: &mut Activation<'_, 'gc, '_>,
+    this: Option<Object<'gc>>,
+    _args: &[Value<'gc>],
+) -> Result<Value<'gc>, Error> {
+    if let Some(bytearray) = this.unwrap().as_bytearray() {
+        return Ok(match bytearray.endian() {
+            Endian::Big => AvmString::new(activation.context.gc_context, "bigEndian").into(),
```

Comment:
`AvmString` accepts direct conversion of `&'static str` without an allocation, you can just use `"bigEndian".into()` without calling `AvmString::new`.

Reply:
There's a handful of other unnecessary `AvmString::new` calls elsewhere in this file - I'm not going to spam you with a review for each one.